syntax = "proto3";

package silenteight.datasource.api.gender;

import "google/api/annotations.proto";
import "silenteight/datasource/api/common/common.proto";

option java_package = "com.silenteight.datasource.api.gender";
option java_outer_classname = "GenderInputProto";
option java_multiple_files = true;

service GenderInputService {
  rpc BatchGetMatchGenderInputsV1 (silenteight.datasource.api.common.BatchGetMatchAgentInputsRequest) returns (stream BatchGetMatchGenderInputsV1Response) {
    option (google.api.http) = {
      post: "/agent-inputs/gender/versions/1/matches:batchGet",
      body: "*"
    };
  }
}

message BatchGetMatchGenderInputsV1Response {
  repeated GenderInput gender_inputs = 1;
}

message GenderInput {
  string match = 1;
  repeated GenderFeatureInput gender_feature_inputs = 2;
}

message GenderFeatureInput {
  string feature = 1;
  repeated string alerted_party_genders = 2;
  repeated string watchlist_genders = 3;
}
